### ISSUE TITLE
Persistent Grid-like Artifacts in Generated Images After Fine-tuning

### ISSUE THREAD

**Issue by nkkbr** [Microtreei/TSD-SR#21]
### **1. Summary**

First of all, thank you for this fantastic project!

I have been fine-tuning the model following the instructions in the README. While the training process runs smoothly, the final model produces images with persistent grid-like artifacts. I've examined checkpoints from various stages of training and noticed that while the artifacts do lessen in the early stages, the improvement slows down significantly later on. Eventually, it seems to stall, and the grid pattern remains a persistent issue that doesn't resolve with more training time.

I'm opening this issue to seek advice on potential causes and to understand if this is a known phenomenon.

#### **2.1. Dataset Preparation**

I prepared my dataset according to the preprocessing steps outlined in the README. The dataset is a combination of standard public datasets and my own custom data.

The composition is as follows:
- **Total Images:** 1,055,920
- **Public Datasets (LSDIR, FLICKR2K, DIV2K, FFHQ):** 286,900 images (approx. 27%)
- **Custom Dataset (high-quality frames from various video sources):** 769,020 images (approx. 73%)

#### **2.2. Training Environment**
- **Hardware:** 4x NVIDIA H100 GPUs

#### **2.3. Training Command**

Here is the exact command I used to launch the fine-tuning process:

```bash
/home/user1/TSD-SR/train/train.py \
--pretrained_model_name_or_path=/home/user1/stable-diffusion-3-medium-diffusers \
--teacher_lora_path=checkpoint/teacher/ \
--train_batch_size=4 \
--rank=64 \
--rank_vae=64 \
--rank_lora=64 \
--num_train_epochs=200 \
--checkpointing_steps=10000 \
--validation_steps=1000 \
--max_train_steps=400000 \
--learning_rate=5e-06 \
--learning_rate_reg=1e-06 \
--lr_scheduler=cosine_with_restarts \
--lr_warmup_steps=3000 \
--seed=43 \
--use_default_prompt \
--use_teacher_lora \
--use_random_bias \
--output_dir=checkpoint/tsdsr-save/ \
--report_to=wandb \
--log_code \
--log_name=tsdsr-train_20250612 \
--gradient_accumulation_steps=1 \
--resume_from_checkpoint=latest \
--guidance_scale=7.5
```

### **3. The Issue: Grid-like Artifacts**

After training for 400,000 steps, the resulting model consistently generates images with a noticeable grid pattern.

Crucially, by inspecting checkpoints saved every 50,000 steps, I found that these artifacts appear relatively early in the training and do not diminish as training progresses. This suggests the issue is not something that resolves with more training time.

### **4. My Questions**

I would be very grateful if you could provide some insight into the following:

1.  **Potential Causes:** What might be the reason for these grid-like artifacts? Is there something I might have overlooked in my data preparation, training parameters, or overall setup?
2.  **Author's Experience:** During your own training and experiments, did you ever encounter similar artifacts? For instance, did they appear early on and then get resolved with further training, or was this issue not present at all in your experience?

Thank you for your time and any help you can provide. I'm looking forward to your feedback

To better illustrate the problem, here are some test examples generated by our fine-tuned model. The grid-like artifacts are clearly visible.

** checkpoint-500 **

![Image](https://github.com/user-attachments/assets/3028a21e-5bd3-4fe5-809c-110f3286ea4b)

** checkpoint-50500 **

![Image](https://github.com/user-attachments/assets/44107bc4-9d19-4ce9-9082-392d66ed7541)

** checkpoint-100500 **

![Image](https://github.com/user-attachments/assets/dc46bff1-c93c-4498-95b3-0cd507ee8cd4)

** checkpoint-150500 **

![Image](https://github.com/user-attachments/assets/308862c5-9ea2-4965-98cc-10c90c051594)

** checkpoint-200500 **

![Image](https://github.com/user-attachments/assets/816a664e-7c0f-4b60-93a0-b2d426253fc0)

** checkpoint-250500 **

![Image](https://github.com/user-attachments/assets/e494479f-a99c-4d4a-a700-7fb8f3198acf)

** checkpoint-300500 **

![Image](https://github.com/user-attachments/assets/8e09196a-ccb9-4967-8e5a-12e985df8713)

** checkpoint-350500 **

![Image](https://github.com/user-attachments/assets/c4113b67-3a3a-4f3b-b09d-c74726f5656e)

** checkpoint-400000 **

![Image](https://github.com/user-attachments/assets/cac20afd-3ff6-4342-990b-c9b524db1e0d)

**Comment by DragonisCV** [comment]
Hi, I have the same problem but still not sure how to deal with it. FluxSR also claims this issue so maybe it will work? btw, I am wondering how fast your training is in this setting, cause it's really really slow ( about 9s /iter)) on A800

**Comment by nkkbr** [comment]
@DragonisCV 

Hi, thank you for your comment!

In my case, with the above setting (including hardware, training script, and dataset size), the training actually took around **212 hours** to complete.

When you mentioned "FluxSR also claims this issue so maybe it will work?", could you clarify what you mean exactly? Are you referring to FluxSR having a potential fix or workaround for this issue?

Thanks!

**Comment by Microtreei** [comment]
Thank you very much for your interest.
We have also found this problem and it will persist in later training if we can't fix it initially. We found that this also happens with lpips alone, probably because the single step of sd3 is harder to train.
[fluxsr](https://arxiv.org/abs/2502.01993) also found this problem on training the flux-based sr method, and they have proposed losses that may solve this problem better.
We will mention a couple of strategies below that may be helpful in solving this problem.

**Comment by Microtreei** [comment]
1. At the beginning of training, it is better to use more diversified prompt descriptions rather than fixed prompt. our fixed prompt training is obtained by fine-tuning at a later stage.
2. The weights of tsm and mse can be tuned up in the early stage. Too large vsd in the early stage will mislead the optimization direction of the model, and the learning rate can be gradually tuned down.
3. When we recently pruned sd3, we found that adding GAN for fine-tuning helps to solve this problem. The way of GAN refers to the way of [ADD](https://arxiv.org/abs/2311.17042), and no need for R1 regularity can also maintain a relatively stable training.
4. You can refer to the loss of fluxsr to try to see if it has improved the problem.

**Comment by DragonisCV** [comment]
> 1. At the beginning of training, it is better to use more diversified prompt descriptions rather than fixed prompt. our fixed prompt training is obtained by fine-tuning at a later stage.
> 2. The weights of tsm and mse can be tuned up in the early stage. Too large vsd in the early stage will mislead the optimization direction of the model, and the learning rate can be gradually tuned down.
> 3. When we recently pruned sd3, we found that adding GAN for fine-tuning helps to solve this problem. The way of GAN refers to the way of [ADD](https://arxiv.org/abs/2311.17042), and no need for R1 regularity can also maintain a relatively stable training.
> 4. You can refer to the loss of fluxsr to try to see if it has improved the problem.

Thanks for your advices!
Given that it's the keypoint of distilling a one-step dit model, could you please release a complete training script that can stably train without artifacts? It might be helpful for us.

**Comment by Microtreei** [comment]
I may be able to provide a strategy for joining GAN training, which is unrelated to this paper, but it is better to solve the artifacts problem in one step. This is how I train when pruning models.

**Comment by nkkbr** [comment]
Hi, @Microtreei , thank you for your detailed response.

I'm not doing research in this area recently— I'm just trying to use the model — but I can really feel how complex and unstable the training process is. It doesn't seem to work simply by following the steps in the repo.

I did follow all the instructions in the repository carefully, including using the same dataset. The only difference is that I also added a large amount of my own data.

Thanks again for your helpful insights, and I wish you the best with your research!

**Comment by Microtreei** [comment]
Thank you for your feedback. I have been reviewing my training process over the past few days and found that there are indeed some tricks that were not included in the final training script (I am testing whether these tricks alleviate the artifacts.). If possible, I will reconfigure the training script over the next few days. Please forgive me for the slow update, as I have been working on another pruning project recently.

Additionally, I will also try to add a version using GAN training, as I found this approach to be effective in addressing this issue during my pruning tasks. Although it is unrelated to this project, I hope this method can provide inspiration for those who need to train single-step models using SD3.

**Comment by ruolunhui** [comment]
Hello, excuse me, I see that the marking has been completed, do you want to ask if it has been solved? I'm also having this problem
